### ISSUE TITLE
Document probabilities doesn't add up to 1.0

### ISSUE THREAD

**Issue by MorFilo** [MaartenGr/BERTopic#500]
The topics and probabilities are not directly saved in the model as especially the probabilities can get quite large. However, you can extract them like this:

```python
# In BERTopic v0.9.2 or higher:
topics= topic_model._map_predictions(topic_model.hdbscan_model.labels_)

probs = hdbscan.all_points_membership_vectors(topic_model.hdbscan_model)
probs = topic_model._map_probabilities(probs, original_topics=True)

# In BERTopic v0.9.2 or lower:
probs = hdbscan.all_points_membership_vectors(topic_model.hdbscan_model)
probs = topic_model._map_probabilities(probs)

topics = [topic_model.mapped_topics[topic] for topic in topic_model.hdbscan_model.labels_]
```

_Originally posted by @MaartenGr in https://github.com/MaartenGr/BERTopic/issues/283#issuecomment-941933954_

I trained my model and extracted the probabilities as mentioned above.
As a part of my task I plan to use those probabilities to measure similarity between different documents.
When summing the probabilities of a document over its  calculated topics I would expect it to sum to 1.0 but its not the case, each document`s topics sums to a different result.
Am I looking at the probabilities in a wrong way ?

**Comment by MaartenGr** [comment]
If I am not mistaken, the sum will be the probability that the point is in any cluster. It is possible that certain points do not belong in any cluster as some points are considered to be outliers. Therefore, the total probability represents whether a point is actually in **any** cluster.

**Comment by MaartenGr** [comment]
Since this was active a while ago, I am closing this for now. Let me know if you want to re-open the issue to continue the discussion.

**Comment by Vijaysinh-DS** [comment]
@MaartenGr I am facing a similar issue when I tried to obtain probabilities for unseen (test) data using `.transform` method but the probability does not sum to 1 and also for training data

**Comment by Vijaysinh-DS** [comment]
@MaartenGr  Thank you for your response!  

i'm Elobaring scenario here for better visibility of the issue I'm facing:

I'm fitting BERTopic() on processed sentences from given PDF file like below:

`  
from sentence_transformers import SentenceTransformer

# Pre-calculate embeddings
embedding_model = SentenceTransformer("BAAI/bge-base-en-v1.5")
embeddings = embedding_model.encode(all_processed_sentences, show_progress_bar=True)

from sklearn.feature_extraction.text import CountVectorizer
vectorizer_model = CountVectorizer(stop_words="english", ngram_range=(1, 1))

from transformers import pipeline
from bertopic.representation import TextGeneration

prompt = """I have a topic that contains the following documents: 
[DOCUMENTS]
The topic is described by the following keywords: [KEYWORDS]

Based on the information above, provide a concise yet descriptive topic label that summarizes the main idea. The label should be a short phrase (3-5 words) that captures the essence of the topic.

Topic label: """

# Create your representation model
generator = pipeline('text2text-generation', model='google/flan-t5-large')
llm = TextGeneration(generator, prompt=prompt)
  

from bertopic.representation import MaximalMarginalRelevance

# MMR
mmr_model = MaximalMarginalRelevance(diversity=0.3)


# All representation models
representation_model = {
    "flan-t5":llm,
    "MMR": mmr_model}


from bertopic import BERTopic
# cluster_model = KMeans(n_clusters=3)
topic_model = BERTopic(

  # Pipeline models
  embedding_model=embedding_model,
  vectorizer_model=vectorizer_model,
  representation_model=representation_model,
  calculate_probabilities=True,
  # Hyperparameters
  top_n_words=10,
  min_topic_size=200, # minimum number of documents in a topic
  verbose=True
)

# Train model
topics, probs = topic_model.fit_transform(all_processed_sentences)
`

Now on top of this fitted model, I'm trying to run similarity over unseen data `topic_model.transform(unseen_data)` which returns topics and probs distribution. Now this distribution is basically for every single document within the collection of `unseen_data`  and if I do sum over non outlier topic (e.g `sum(probs[2])`) it doesn't add up to 1, but it should sum up to 1 right? 

Ultimately I'm trying to obtain similarity score (how similar is the unseen data to training data) which ideally should sum up to 1  

Tried minimizing the outliers by [outlier reduction techniques](https://maartengr.github.io/BERTopic/getting_started/outlier_reduction/outlier_reduction.html) and even changed clustering  algorithm, the results were still same. 

Is this approach correct? or i'm missing any configurations?

**Comment by MaartenGr** [comment]
> Now this distribution is basically for every single document within the collection of unseen_data and if I do sum over non outlier topic (e.g sum(probs[2])) it doesn't add up to 1, but it should sum up to 1 right?

No, it should not. The reason for this is that the probabilities that you get in `probs` are all topics **except** the outlier topic (that one is currently missing as it is HDBSCAN specific behavior). Therefore, if you want it to sum to 1 you need to add the outlier probability which is 1 - sum(probs). 

If you do not want to include the outlier probability, then you could do a softmax instead and get them to 1.
